--- tools/maspsx/maspsx/__init__.py	2026-09-30 00:30:06.667055200 -0500
+++ tools/maspsx/maspsx/__init__.py	2026-09-30 12:01:15.251093400 -0500
@@ -636,7 +636,7 @@
                 # e.g.	.comm	MENU_RadarScale_800AB480,4
                 in_sdata = False
                 _, var = line.split()
-                symbol, size_str = var.split(",")
+                symbol, size_str = var.split(",")[:2]
                 size = int(size_str)
                 if size <= self.sdata_limit:
                     self.sbss_entries[symbol] = size
