--- j0
+++ b1_proto_copy_svector.c
@@ -926,19 +926,18 @@
 }
 void func_80049710(void) {
 }
-typedef struct { s32 f0, f1, f2, f3, f4, f5, f6, f7; } _struct_copy_func49718;
 
 extern u8 *D_800A38B4;
 extern s16 D_800EF980[];
 extern s32 (*g_anim_func_table[])(s16 *, s16 *);
 
 extern void func_80052C10(void);
-extern void MulMatrix0(s16 *, s16 *, s16 *);
+extern MATRIX *MulMatrix0(MATRIX *, MATRIX *, MATRIX *);
 extern VECTOR *ApplyMatrix(MATRIX *, SVECTOR *, VECTOR *);
 
 void func_80049718(s32 arg0, s32 arg1, s32 *arg2, s16 *arg3) {
     int new_var2;
-    s16 sp10[3];
+    SVECTOR rot;
     s16 *p_anim;
     s32 var_s5;
     s32 var_s3;
@@ -985,16 +984,16 @@
             *((s32 *) (part + 0x50)) = ((*((s32 *) (part + 0x50))) * (*((s16 *) (vehicle + 0x12)))) >> 12;
             *((s32 *) (part + 0x54)) = ((*((s32 *) (part + 0x54))) * (*((s16 *) (vehicle + 0x12)))) >> 12;
             p_anim = vehicle + 0x44;
-            MulMatrix0((s16 *) p_anim, (s16 *) (part + 0x38), (s16 *) (obj + 0x18));
-            sp10[0] = (s16) (*((s32 *) (part + 0x4C)));
-            sp10[1] = (s16) (*((s32 *) (part + 0x50)));
-            sp10[2] = (s16) (*((s32 *) (part + 0x54)));
-            ApplyMatrix((MATRIX *)(*((s32 *) (part + 0xC)) + 0x18), (SVECTOR *)sp10, (VECTOR *) (obj + 0x2C));
+            MulMatrix0((MATRIX *) p_anim, (MATRIX *) (part + 0x38), (MATRIX *) (obj + 0x18));
+            rot.vx = *((s32 *) (part + 0x4C));
+            rot.vy = *((s32 *) (part + 0x50));
+            rot.vz = *((s32 *) (part + 0x54));
+            ApplyMatrix((MATRIX *)(*((s32 *) (part + 0xC)) + 0x18), &rot, (VECTOR *) (obj + 0x2C));
             *((s32 *) (obj + 0x2C)) = (*((s32 *) (obj + 0x2C))) + (*((s32 *) ((*((u8 **) (part + 0xC))) + 0x2C)));
             *((s32 *) (obj + 0x30)) = (*((s32 *) (obj + 0x30))) + (*((s32 *) ((*((u8 **) (part + 0xC))) + 0x30)));
             *((s32 *) (obj + 0x34)) = (*((s32 *) (obj + 0x34))) + (*((s32 *) ((*((u8 **) (part + 0xC))) + 0x34)));
             var_s3 = var_s3 | 0x8000;
-            *((_struct_copy_func49718 *) (part + 0x18)) = *((_struct_copy_func49718 *) (obj + 0x18));
+            *(MATRIX *)(part + 0x18) = *(MATRIX *)(obj + 0x18);
             var_s5 = *((s16 *) (vehicle + 0x1A84));
         }
         ot = D_800A3820;
