--- tmp/f23f08b/v_nor.c	2026-10-02 01:59:13.812159100 -0500
+++ tmp/f23f08b/v_range.c	2026-10-02 01:59:13.812159100 -0500
@@ -571,7 +571,7 @@
         goto set_8c;
     }
     if (((rec->unk_0C == 0x1D || rec->unk_0C == 0xE) && (rec->unk_6A == 2 || rec->unk_6A == 0x1B || rec->unk_6A == 0x28 || rec->unk_6A == 0x26) && rec->unk_A1[1] < 0xFF && rec->unk_26C != 0 && rec->unk_40 >= rec->unk_A5 && rec->unk_40 <= rec->unk_A6)
-        || (rec->unk_0A == 0xE && rec->unk_6A == 0x11 && D_800A38AE == arg0 && rec->unk_40 >= rec->unk_A5 && rec->unk_A6 >= rec->unk_40)) {
+        || (rec->unk_0A == 0xE && rec->unk_6A == 0x11 && D_800A38AE == arg0 && rec->unk_40 >= rec->unk_A5 && rec->unk_40 <= rec->unk_A6)) {
     set_8c:
         rec->unk_8C = rec->unk_8C != 0 ? 1 : 2;
         rec->unk_62 |= 4;
