--- target
+++ cc1psx
@@ -20,3 +20,3 @@
 lw v0,4(v1)
-lw s1,<D_800A37D4>(gp)
+nop
 lhu v0,2(v0)
@@ -25,3 +25,3 @@
 lw v0,4(v1)
-lw a1,<D_800A3470>(gp)
+nop
 lhu v0,4(v0)
@@ -30,2 +30,4 @@
 lw v0,8(v1)
+lw a1,<D_800A3470>(gp)
+lw s1,<D_800A37D4>(gp)
 lw a2,<D_800A3474>(gp)
@@ -35,3 +37,3 @@
 lw v0,8(v1)
-lw s0,<D_800A34EC>(gp)
+nop
 lw v0,4(v0)
@@ -40,2 +42,3 @@
 lw v1,8(v1)
+lw s0,<D_800A34EC>(gp)
 lw v0,<D_800A32B8>(gp)
@@ -53,11 +56,9 @@
 sh zero,52(s0)
+lw v1,<D_800A34B0>(gp)
 sll v0,v0,8
 addiu s3,s0,16
+sw v0,0(v1)
+lw v1,<D_800A3490>(gp)
 addiu s2,s0,18
 addiu s7,s0,20
-addiu s4,s0,36
-lw v1,<D_800A34B0>(gp)
-addiu t0,s0,52
-sw v0,0(v1)
-lw v1,<D_800A3490>(gp)
 addiu v0,zero,47
@@ -66,2 +67,4 @@
 lw v0,<lo:0x800f1198>(v0)
+addiu s4,s0,36
+addiu t0,s0,52
 addiu s6,s0,68
@@ -70,3 +73,3 @@
 andi v0,v0,1
-beqz v0,L
+beq v0,zero,L
 sw t0,24(sp)
@@ -88,5 +91,5 @@
 slti v0,v1,2
-beqz v0,L
-nop
-beqz v1,L
+beq v0,zero,L
+nop
+beq v1,zero,L
 addiu v0,zero,450
@@ -113,4 +116,4 @@
 lw a0,<D_800A32B8>(gp)
-lui v0,(2863311531 >> 16)
-ori v0,v0,(2863311531 & 65535)
+lui v0,43690
+ori v0,v0,43691
 multu a0,v0
@@ -121,6 +124,6 @@
 sll v0,v0,1
+lui v1,<hi:0x800a>
+lw v1,<lo:0x8009bd44>(v1)
 subu a0,a0,v0
 sll a0,a0,3
-lui v0,<hi:0x800a>
-lw v0,<lo:0x8009bd44>(v0)
 addu a0,a0,fp
@@ -128,4 +131,4 @@
 sw a0,<D_800A348C>(gp)
-andi v0,v0,1
-beqz v0,L
+andi v1,v1,1
+beq v1,zero,L
 nop
@@ -156,4 +159,2 @@
 sh v0,0(v1)
-lui t0,<hi:0x800a>
-addiu t0,t0,<lo:0x8009ba30>
 lw v0,<D_800A32B8>(gp)
@@ -161,2 +162,4 @@
 lw v1,<lo:0x8009bd44>(v1)
+lui t0,<hi:0x800a>
+addiu t0,t0,<lo:0x8009ba30>
 andi v0,v0,3
@@ -167,3 +170,3 @@
 sw v0,<D_800A348C>(gp)
-beqz v1,L
+beq v1,zero,L
 nop
@@ -186,15 +189,15 @@
 addu v0,v0,v1
-sll a2,v0,2
+sll a1,v0,2
+lw a2,<D_800A3470>(gp)
 lui at,<hi:0x800f>
-addu at,at,a2
+addu at,at,a1
 lw v0,<lo:0x800f1198>(at)
-lw a1,<D_800A3470>(gp)
+lw a0,0(a2)
 srl v1,v0,31
 addu v0,v0,v1
-lw v1,0(a1)
 sra v0,v0,1
-subu v0,v0,v1
+subu v0,v0,a0
 sw v0,0(s4)
 lui at,<hi:0x800f>
-addu at,at,a2
+addu at,at,a1
 lw v0,<lo:0x800f119c>(at)
@@ -204,4 +207,3 @@
 addiu v0,v0,7
-addu a0,s4,zero
-lw v1,4(a1)
+lw v1,4(a2)
 sra v0,v0,3
@@ -210,5 +212,6 @@
 lui at,<hi:0x800f>
-addu at,at,a2
+addu at,at,a1
 lw v0,<lo:0x800f11a0>(at)
-lw v1,8(a1)
+lw v1,8(a2)
+addu a0,s4,zero
 addu a1,s7,zero
@@ -219,5 +222,5 @@
 addiu a0,s7,-20
-lw a0,16(sp)
 lw a1,<D_800A34B8>(gp)
 lw a3,<D_800A34CC>(gp)
+lw a0,16(sp)
 lw a2,24(sp)
@@ -227,3 +230,3 @@
 nop
-swc2 19,0(t0)
+swc2 $19,0(t0)
 lw v0,<D_800A34D0>(gp)
@@ -235,3 +238,3 @@
 sltiu v0,v1,4101
-beqz v0,L
+beq v0,zero,L
 sw v1,0(s6)
@@ -246,3 +249,3 @@
 sltu v0,v0,v1
-beqz v0,L
+beq v0,zero,L
 addiu a0,s5,1
@@ -252,2 +255,3 @@
 lhu v1,2(v1)
+lw a0,<D_800A3494>(gp)
 srl v0,v0,4
@@ -256,5 +260,4 @@
 addu v0,v0,v1
-lw v1,<D_800A3494>(gp)
 andi v0,v0,65535
-sw v0,0(v1)
+sw v0,0(a0)
 lw v0,<D_800A3488>(gp)
@@ -273,3 +276,3 @@
 nop
-bnez v0,L
+bne v0,zero,L
 nop
@@ -283,11 +286,11 @@
 div zero,v1,v0
-bnez v0,L
-nop
-break 7
+bne v0,zero,L
+nop
+break 7,L
 addiu at,zero,-1
 bne v0,at,L
-lui at,(2147483648 >> 16)
+lui at,32768
 bne v1,at,L
 nop
-break 6
+break 6,L
 mflo v1
@@ -304,3 +307,3 @@
 slti v0,a0,513
-bnez v0,L
+bne v0,zero,L
 addiu v0,zero,2
@@ -321,3 +324,3 @@
 slti v0,a0,513
-bnez v0,L
+bne v0,zero,L
 addiu v0,zero,2
@@ -330,4 +333,4 @@
 mflo v0
-lui v1,(1717986919 >> 16)
-ori v1,v1,(1717986919 & 65535)
+lui v1,26214
+ori v1,v1,26215
 mult v0,v1
@@ -425,4 +428,5 @@
 lw v0,<D_800A34A4>(gp)
+nop
+lhu v0,0(v0)
 addiu a1,zero,1
-lhu v0,0(v0)
 addu a0,s1,zero
@@ -447,29 +451,29 @@
 slti v0,v0,449
-beqz v0,L
+beq v0,zero,L
 addiu a0,s5,1
-lui a2,(16777215 >> 16)
-ori a2,a2,(16777215 & 65535)
+lw v0,0(s6)
+lui v1,<hi:0x800a>
+lw v1,<lo:0x800a374c>(v1)
+lui a2,255
+sw s1,<D_800A34E8>(gp)
+sll v0,v0,2
+addu v1,v1,v0
+sw v1,<D_800A34E4>(gp)
+lw v0,0(s1)
+lw v1,0(v1)
+ori a2,a2,65535
+lui a1,65280
+and v0,v0,a1
+and v1,v1,a2
+or v0,v0,v1
+sw v0,0(s1)
+lw a0,<D_800A34E4>(gp)
+lw v0,<D_800A34E8>(gp)
+lw v1,0(a0)
 addiu s0,s0,40
-lw v1,0(s6)
-lui v0,<hi:0x800a>
-lw v0,<lo:0x800a374c>(v0)
-lui a1,(4278190080 >> 16)
-sw s1,<D_800A34E8>(gp)
-sll v1,v1,2
-addu v0,v0,v1
-sw v0,<D_800A34E4>(gp)
-lw v1,0(s1)
-lw v0,0(v0)
+addiu s1,s1,40
+and v0,v0,a2
 and v1,v1,a1
-and v0,v0,a2
-or v1,v1,v0
-sw v1,0(s1)
-addiu s1,s1,40
-lw a0,<D_800A34E4>(gp)
-lw v1,<D_800A34E8>(gp)
-lw v0,0(a0)
-and v1,v1,a2
-and v0,v0,a1
-or v1,v1,v0
-sw v1,0(a0)
+or v0,v0,v1
+sw v0,0(a0)
 addiu a0,s5,1
@@ -485,3 +489,3 @@
 andi v0,v0,1
-bnez v0,L
+bne v0,zero,L
 addu s5,a0,zero
