--- src/text1b.c
+++ v1_scalar_direct.c
@@ -661,8 +661,6 @@
     s32 index;
     s32 scale;
     s32 old;
-    s16 *rotp;
-    s32 *vec;
     s16 *indices;
     s32 *ot;
     u8 *player;
@@ -683,10 +681,8 @@
     rot.vz = ((s32)(&Judge)[arg0 & 0xFFF] * scale) >> 12;
     rot.vx = ((s32)(&Judge)[(arg0 + 0x400) & 0xFFF] * scale) >> 12;
     prim = (u8 *)D_800A33E4;
-    rotp = &rot.vx;
-    vec = &D_800FF56C;
-    ApplyMatrix(*(s32 *)player, rotp, vec);
-    vec[0] += *(s32 *)(*(u8 **)player + 0x14);
+    ApplyMatrix(*(s32 *)player, &rot.vx, &D_800FF56C);
+    D_800FF56C += *(s32 *)(*(u8 **)player + 0x14);
     D_800FF570 += *(s32 *)(*(u8 **)(player + 4) + 0x18);
     D_800FF574 += *(s32 *)(*(u8 **)(player + 8) + 0x1C);
 
@@ -694,7 +690,7 @@
     rot.vy = 0xC00 - arg0;
     rot.vz = 0;
     indices = D_80099C14;
-    math_RotMatrixZYX(rotp, mtx.m[0]);
+    math_RotMatrixZYX(&rot.vx, mtx.m[0]);
     gte_MulMatrix0ClearTrans(*(s32 *)player, mtx.m[0], mtx.m[0]);
     D_800FF558 = mtx.m[0][0];
     D_800FF55A = mtx.m[1][0];
