--- src/text1b.c
+++ v2b_matrix_addr_t.c
@@ -632,9 +632,9 @@
 typedef struct { s32 f0, f1, f2, f3, f4, f5, f6, f7; } _struct_copy_func48BA4;
 extern void *game_GetPlayerData();
 extern s16 Judge;
-extern void ApplyMatrix(s32, s16 *, s32 *);
-extern void math_RotMatrixZYX(s16 *, s16 *);
-extern void gte_MulMatrix0ClearTrans(s32, s16 *, s16 *);
+extern void ApplyMatrix(MATRIX *, SVECTOR *, VECTOR *);
+extern void math_RotMatrixZYX(SVECTOR *, MATRIX *);
+extern void gte_MulMatrix0ClearTrans(MATRIX *, MATRIX *, MATRIX *);
 extern s32 ClearOTagR(s32, s32);
 extern s32 D_800A36AC;
 extern s32 D_800A378C;
@@ -642,18 +642,7 @@
 extern s32 g_gpu_ot256_ptr;
 extern u8 g_gpu_ot256_db[];
 extern s16 D_80099C14[];
-extern s16 D_800FF558;
-extern s16 D_800FF55A;
-extern s16 D_800FF55C;
-extern s16 D_800FF55E;
-extern s16 D_800FF560;
-extern s16 D_800FF562;
-extern s16 D_800FF564;
-extern s16 D_800FF566;
-extern s16 D_800FF568;
-extern s32 D_800FF56C;
-extern s32 D_800FF570;
-extern s32 D_800FF574;
+extern MATRIX D_800FF558;
 
 void func_80048BA4(s32 arg0, s32 arg1, s32 arg2) {
     MATRIX mtx;
@@ -661,8 +650,6 @@
     s32 index;
     s32 scale;
     s32 old;
-    s16 *rotp;
-    s32 *vec;
     s16 *indices;
     s32 *ot;
     u8 *player;
@@ -683,28 +670,26 @@
     rot.vz = ((s32)(&Judge)[arg0 & 0xFFF] * scale) >> 12;
     rot.vx = ((s32)(&Judge)[(arg0 + 0x400) & 0xFFF] * scale) >> 12;
     prim = (u8 *)D_800A33E4;
-    rotp = &rot.vx;
-    vec = &D_800FF56C;
-    ApplyMatrix(*(s32 *)player, rotp, vec);
-    vec[0] += *(s32 *)(*(u8 **)player + 0x14);
-    D_800FF570 += *(s32 *)(*(u8 **)(player + 4) + 0x18);
-    D_800FF574 += *(s32 *)(*(u8 **)(player + 8) + 0x1C);
+    ApplyMatrix(*(MATRIX **)player, &rot, (VECTOR *)&D_800FF558.t);
+    D_800FF558.t[0] += (*(MATRIX **)player)->t[0];
+    D_800FF558.t[1] += ((MATRIX **)player)[1]->t[1];
+    D_800FF558.t[2] += ((MATRIX **)player)[2]->t[2];
 
     rot.vx = 0;
     rot.vy = 0xC00 - arg0;
     rot.vz = 0;
     indices = D_80099C14;
-    math_RotMatrixZYX(rotp, mtx.m[0]);
-    gte_MulMatrix0ClearTrans(*(s32 *)player, mtx.m[0], mtx.m[0]);
-    D_800FF558 = mtx.m[0][0];
-    D_800FF55A = mtx.m[1][0];
-    D_800FF55C = mtx.m[2][0];
-    D_800FF55E = mtx.m[0][1];
-    D_800FF560 = mtx.m[1][1];
-    D_800FF562 = mtx.m[2][1];
-    D_800FF564 = mtx.m[0][2];
-    D_800FF566 = mtx.m[1][2];
-    D_800FF568 = mtx.m[2][2];
+    math_RotMatrixZYX(&rot, &mtx);
+    gte_MulMatrix0ClearTrans(*(MATRIX **)player, &mtx, &mtx);
+    D_800FF558.m[0][0] = mtx.m[0][0];
+    D_800FF558.m[0][1] = mtx.m[1][0];
+    D_800FF558.m[0][2] = mtx.m[2][0];
+    D_800FF558.m[1][0] = mtx.m[0][1];
+    D_800FF558.m[1][1] = mtx.m[1][1];
+    D_800FF558.m[1][2] = mtx.m[2][1];
+    D_800FF558.m[2][0] = mtx.m[0][2];
+    D_800FF558.m[2][1] = mtx.m[1][2];
+    D_800FF558.m[2][2] = mtx.m[2][2];
 
     goto test_index;
 copy_index:
@@ -954,7 +939,6 @@
 
 extern void func_80052C10(void);
 extern void MulMatrix0(s16 *, s16 *, s16 *);
-extern void ApplyMatrix(s32, s16 *, s32 *);
 
 void func_80049718(s32 arg0, s32 arg1, s32 *arg2, s16 *arg3) {
     int new_var2;
@@ -1009,7 +993,7 @@
             sp10[0] = (s16) (*((s32 *) (part + 0x4C)));
             sp10[1] = (s16) (*((s32 *) (part + 0x50)));
             sp10[2] = (s16) (*((s32 *) (part + 0x54)));
-            ApplyMatrix((*((s32 *) (part + 0xC))) + 0x18, sp10, (s32 *) (obj + 0x2C));
+            ApplyMatrix((MATRIX *)(*((s32 *) (part + 0xC)) + 0x18), (SVECTOR *)sp10, (VECTOR *) (obj + 0x2C));
             *((s32 *) (obj + 0x2C)) = (*((s32 *) (obj + 0x2C))) + (*((s32 *) ((*((u8 **) (part + 0xC))) + 0x2C)));
             *((s32 *) (obj + 0x30)) = (*((s32 *) (obj + 0x30))) + (*((s32 *) ((*((u8 **) (part + 0xC))) + 0x30)));
             *((s32 *) (obj + 0x34)) = (*((s32 *) (obj + 0x34))) + (*((s32 *) ((*((u8 **) (part + 0xC))) + 0x34)));
